src/inmanta/agent/in_process_executor.py

```diff
@@ -309,8 +320,17 @@ async def dry_run(
                         )
                     else:
                         try:
+
+                            def resolve_and_dryrun(
+                                provider: HandlerAPI[Resource],
+                                ctx: handler.HandlerContext,
+                                resource: Resource,
+                            ) -> None:
+                                resource.resolve_all_references(ctx)
```

Comment:
@bartv failure added back here  (and in 2 other places)

Reply:
Yeah, but that results in Failure instead of Unavailable? That was actually my question.

Reply:
added testcase to that effect